Mathlib/Algebra/Star/CentroidHom.lean

```diff
@@ -0,0 +1,136 @@
+  star_add f g := SetCoe.ext (star_add f.val g.val)
+
+instance : StarRing (Subsemiring.center (CentroidHom α)) where
+  __ := instStarAddMonoidCenter
+  star_mul f g := by
+    ext a
+    calc
+      star (f * g) a = star (g * f) a := by rw [CommMonoid.mul_comm f g]
+      _ = star (g (f (star a))) := rfl
+      _ = star (g (star (star (f (star a))))) := by simp only [star_star]
+      _ = (star g * star f) a := rfl
+
+/-- The canonical *-homomorphism embedding the center of `CentroidHom α` into `CentroidHom α`. -/
+def centerStarEmbedding : Subsemiring.center (CentroidHom α) →⋆ₙ+* CentroidHom α where
+  toNonUnitalRingHom :=
+    (SubsemiringClass.subtype (Subsemiring.center (CentroidHom α))).toNonUnitalRingHom
+  map_star' f := by
+    simp only [RingHom.toMonoidHom_eq_coe, OneHom.toFun_eq_coe, MonoidHom.toOneHom_coe,
+      MonoidHom.coe_coe, SubsemiringClass.coe_subtype]
+    exact rfl
+
+theorem inr_star (z : NonUnitalStarSubsemiring.center α) :
+    star (centerToCentroidCenter z) =
+      (centerToCentroidCenter (star z : NonUnitalStarSubsemiring.center α)) := by
+  ext a
+  calc
+      (star (centerToCentroidCenter z)) a = star (z * star a) := rfl
+      _ = star (star a) * star z := by simp only [star_mul, star_star, StarMemClass.coe_star]
+      _ = a * star z := by rw [star_star]
+      _ = (star z) * a := by rw [(star z).property.comm]
+      _ = (centerToCentroidCenter ((star z) : NonUnitalStarSubsemiring.center α)) a := rfl
+
+/-- The canonical *-homomorphism from the center of a non-unital, non-associative *-semiring into
+the center of its centroid. -/
+def starCenterToCentroidCenter :
+    NonUnitalStarSubsemiring.center α →⋆ₙ+* Subsemiring.center (CentroidHom α) where
+  toNonUnitalRingHom := centerToCentroidCenter
+  map_star' _ := by
+    simp only [MulHom.toFun_eq_coe, NonUnitalRingHom.coe_toMulHom]
+    exact (inr_star _).symm
+
+/-- The canonical homomorphism from the center into the centroid -/
+def starCenterToCentroid : NonUnitalStarSubsemiring.center α →⋆ₙ+* CentroidHom α :=
+  NonUnitalStarRingHom.comp (centerStarEmbedding) (starCenterToCentroidCenter)
+
+lemma starCenterToCentroid_apply (z : NonUnitalStarSubsemiring.center α) (a : α) :
+    (starCenterToCentroid z) a = z * a := rfl
+
+/--
+Let `α` be a star ring with commutative centroid. Then the centroid is a star ring.
+-/
+@[reducible]
+def starRingOfCommCentroidHom (mul_comm : Std.Commutative (α := CentroidHom α) (· * ·)) :
+    StarRing (CentroidHom α) where
+  __ := instStarAddMonoid
+  star_mul _ _ := ext (fun _ => by
+    rw [mul_comm.comm, star_apply, mul_apply, mul_apply, star_apply, star_apply, star_star])
+
+end NonUnitalNonAssocStarSemiring
+
+section NonAssocStarSemiring
+
+variable [NonAssocSemiring α] [StarRing α]
+
+/-- The canonical isomorphism from the center of a (non-associative) semiring onto its centroid. -/
+def starCenterIsoCentroid : StarSubsemiring.center α ≃⋆+* CentroidHom α where
```

Comment:
Could you add `simp` lemmas for this too? `simps` generates `_symm_apply_coe` correctly, but leaves a stray `toFun` in the other lemma.

(As a reminder, you can check what `simps` does with `@[simps?]`)

Reply:
Like this? https://github.com/leanprover-community/mathlib4/pull/6595/commits/6f262c709d8dc083bb4685a085d1ae7159021052